api/client/src/main/java/org/projectnessie/client/auth/AwsAuthenticationProvider.java

```diff
@@ -85,6 +84,8 @@ private static Region awsRegion(String regionName) {
   private static class AwsAuthentication implements HttpAuthentication {
     private final Region region;
     private final AwsCredentialsProvider awsCredentialsProvider;
+    private final ObjectMapper objectMapper;
+    private final Aws4Signer signer;
```

Comment:
I think we do not need to memoize the `signer` instance, the implementations don't keep any state.